plan/aggregation_push_down.go

```diff
@@ -197,13 +197,19 @@ func (a *aggPushDownSolver) allFirstRow(aggFuncs []expression.AggregationFunctio
 }
 
 // tryToPushDownAgg tries to push down an aggregate function into a join path. If all aggFuncs are first row, we won't
-// process it temporarily. If not, We will add additional group by columns and first row functions. We make a new aggregation
-// operator.
+// process it temporarily. If not, We will add additional group by columns and first row functions. We make a new aggregation operator.
+// If the pushed aggregation is grouped by unique key, it's no need to push it down.
 func (a *aggPushDownSolver) tryToPushDownAgg(aggFuncs []expression.AggregationFunction, gbyCols []*expression.Column, join *Join, childIdx int) LogicalPlan {
 	child := join.children[childIdx].(LogicalPlan)
 	if a.allFirstRow(aggFuncs) {
 		return child
 	}
+	tmpSchema := expression.NewSchema(gbyCols...)
+	for _, key := range child.Schema().Keys {
+		if tmpSchema.ColumnsIndices(key) != nil {
```

Comment:
If group by key is `k1, k2` and only 'k1' is unique key, we should handle this case.
You can make gbyCols as `KeyInfo` and check child.Schema().Keys one by one.